File: src/LeetCode/Algorithms/Two_Sum_II_Input_array_is_sorted.java
package LeetCode.Algorithms;

import java.util.HashMap;

// https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/
public class Two_Sum_II_Input_array_is_sorted {
	public static void main(String[] args) {
		int[] numbers = {5, 25, 75}; // {2, 7, 11, 15}; // {-5, -4, -3, -2, -1};
		int target = 100;

		int[] result = twoSum(numbers, target);

		for(int x : result) {
			System.out.print(x + " ");
		}
	}
	
	public static int[] twoSum(int[] numbers, int target) {
		int left = 0;
		int right = numbers.length - 1;
		
		while (left < right) {
			int sum = numbers[left] + numbers[right];
			
			if (sum == target) {
				return new int[]{left + 1, right + 1};
			}
			
			if (sum > target) {
				right--;
			} else {
				left++;
			}
		}
		
		return new int[]{};
	}
}

File: src/LeetCode/Algorithms/JavaScript/Two_Sum_II - Input_array_is_sorted.js
/**
 * https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(numbers, target) {
	const obj = {};
	
	for (let i = 0; i < numbers.length; i++) {
		let key = target - numbers[i];
		
		if (obj[key] !== undefined) {
			return [obj[key] + 1, i + 1];
		}
		
		obj[numbers[i]] = i;
	}
	
	/*const obj = new Map();
	
	for (let i = 0; i < nums.length; i++) {
		let key = target - nums[i];
		
		if (obj.has(key)) {
			return [obj.get(key), i];
		}
		
		obj.set(nums[i], i);
	}*/
};

let numbers = [2, 7, 11, 15];
let target = 9;

console.log(twoSum(numbers, target));